### ISSUE TITLE
Compile errors on Arch

### ISSUE THREAD

**Issue by Zethson** [genometools/genometools#916]
## Problem description
Doesn't compile.

```
[compile ltr_cluster_prepare_seq_visitor.o]
src/ltr/ltr_cluster_prepare_seq_visitor.c: In function ‘gt_ltr_cluster_prepare_seq_visitor_feature_node’:
src/ltr/ltr_cluster_prepare_seq_visitor.c:146:39: error: ‘%lu’ directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  146 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |                                       ^~~~~
In file included from /home/zeth/genometools/src/core/array_api.h:23,
                 from /home/zeth/genometools/src/core/array.h:21,
                 from src/ltr/ltr_cluster_prepare_seq_visitor.c:19:
/home/zeth/genometools/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_prepare_seq_visitor.c:146:14: note: ‘snprintf’ output between 5 and 8234 bytes into a destination of size 8192
  146 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  147 |                       range.end);
      |                       ~~~~~~~~~~
src/ltr/ltr_cluster_prepare_seq_visitor.c:109:39: error: ‘%lu’ directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  109 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |                                       ^~~~~
In file included from /home/zeth/genometools/src/core/array_api.h:23,
                 from /home/zeth/genometools/src/core/array.h:21,
                 from src/ltr/ltr_cluster_prepare_seq_visitor.c:19:
/home/zeth/genometools/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_prepare_seq_visitor.c:109:14: note: ‘snprintf’ output between 5 and 8234 bytes into a destination of size 8192
  109 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  110 |                       range.end);
      |                       ~~~~~~~~~~
[compile ltr_cluster_stream.o]
cc1: all warnings being treated as errors
make: *** [Makefile:825: obj/src/ltr/ltr_cluster_prepare_seq_visitor.o] Error 1
make: *** Waiting for unfinished jobs....
src/ltr/ltr_cluster_stream.c: In function ‘process_feature’:
src/ltr/ltr_cluster_stream.c:352:41: error: ‘%lu’ directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  352 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                                         ^~~~~
In file included from /home/zeth/genometools/src/core/array_api.h:23,
                 from /home/zeth/genometools/src/core/array.h:21,
                 from src/ltr/ltr_cluster_stream.c:19:
/home/zeth/genometools/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_stream.c:352:16: note: ‘snprintf’ output between 5 and 8234 bytes into a destination of size 8192
  352 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  353 |                         range.start, range.end);
      |                         ~~~~~~~~~~~~~~~~~~~~~~~
src/ltr/ltr_cluster_stream.c:344:41: error: ‘%lu’ directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  344 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                                         ^~~~~
In file included from /home/zeth/genometools/src/core/array_api.h:23,
                 from /home/zeth/genometools/src/core/array.h:21,
                 from src/ltr/ltr_cluster_stream.c:19:
/home/zeth/genometools/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_stream.c:344:16: note: ‘snprintf’ output between 5 and 8234 bytes into a destination of size 8192
  344 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  345 |                         range.start, range.end);
      |                         ~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:822: obj/src/ltr/ltr_cluster_stream.o] Error 1

```

## Exact command line call triggering the problem

```
make -j4
```

## What operating system (e.g. Ubuntu, Mac OS X), OS version (e.g. 15.10, 10.11) and platform (e.g. x86_64) are you using?
Arch Linux most recent

**Comment by satta** [comment]
This is caused by a new gcc doing more extensive static checks, same issue as #915.

**Comment by gordon** [comment]
Fixed by #917.
